util/lru/lru_test.go

```diff
@@ -48,6 +49,156 @@ func TestLRU(t *testing.T) {
 	}
 }
 
+func TestLRUDeleteCorruption(t *testing.T) {
+	// Regression test for tailscale/corp#14747
+
+	c := Cache[int, bool]{}
+
+	c.Set(1, true)
+	c.Set(2, true)
```

Comment:
c.Set(2, true) // now 2 is the head
c.Delete(2) // delete the head

Reply:
Done